core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorFactory.java

```diff
@@ -276,18 +280,24 @@ Aggregator create(String name,
 
                 }
                 final IncludeExclude.OrdinalsFilter filter = includeExclude == null ? null : includeExclude.convertToOrdinalsFilter(format);
-                boolean remapGlobalOrds = true;
-                if (includeExclude == null &&
-                        Aggregator.descendsFromBucketAggregator(parent) == false &&
-                        (factories == AggregatorFactories.EMPTY ||
-                            (isAggregationSort(order) == false && subAggCollectMode == SubAggCollectionMode.BREADTH_FIRST))) {
-                    /**
-                     * We don't need to remap global ords iff this aggregator:
-                     *    - has no include/exclude rules AND
-                     *    - is not a child of a bucket aggregator AND
-                     *    - has no sub-aggregator or only sub-aggregator that can be deferred ({@link SubAggCollectionMode#BREADTH_FIRST}).
-                     **/
-                     remapGlobalOrds = false;
+                boolean remapGlobalOrds;
+                if (REMAP_GLOBAL_ORDS != null) {
+                    // We use REMAP_GLOBAL_ORDS to allow tests to force specific optimizations
+                    remapGlobalOrds = REMAP_GLOBAL_ORDS.booleanValue();
```

Comment:
We should always remap if the agg is a child of a bucket agg, even for testing this is too costly to keep all global ords in each bucket.

Reply:
We had a chat with @jpountz , since this booleans are used only in specific tests where the number of unique terms is low this should not be a problem. +1 to keep it like this.